Makefile

```diff
@@ -0,0 +1,139 @@
+ifeq ($(ARCH),ppc64le)
+	CC=powerpc64le-linux-gnu-gcc
+	IMAGE_SUFFIX=-$(ARCH)
+endif
+GOARM=6
+KUBE_CROSS_TAG=v1.6.2-2
+
+dist/flanneld: $(shell find . -type f  -name '*.go')
+	go build -o dist/flanneld \
+	  -ldflags "-extldflags -static -X github.com/coreos/flannel/version.Version=$(TAG)"
+
+test: license-check gofmt
+	go test -cover $(TEST_PACKAGES_EXPANDED)
+
+cover:
+	# A single package must be given - e.g. 'PACKAGES=pkg/ip make cover'
+	go test -coverprofile cover.out $(PACKAGES_EXPANDED)
+	go tool cover -html=cover.out
+
+# Throw an error if gofmt finds problems.
+# "read" will return a failure return code if there is no output. This is inverted wth the "!"
+gofmt:
+	bash -c '! gofmt -d $(PACKAGES) 2>&1 | read'
+
+gofmt-fix:
+	gofmt -w $(PACKAGES)
+
+license-check:
+	./license-check.sh
+
+clean:
+	rm -f dist/flanneld*
+	rm -f dist/iptables*
+	rm -f dist/*.aci
+	rm -f dist/*.docker
+	rm -f dist/*.tar.gz
+
+## Create a docker image on disk for a specific arch and tag
+dist/flanneld-$(ARCH)-$(TAG).docker: dist/flanneld-$(ARCH)
+	docker build -f Dockerfile.$(ARCH) -t $(REGISTRY)/flannel$(IMAGE_SUFFIX):$(TAG) .
+	docker save -o dist/flanneld-$(ARCH)-$(TAG).docker $(REGISTRY)/flannel$(IMAGE_SUFFIX):$(TAG)
+
+## Create an ACI on disk for a specific arch and tag
+dist/flanneld-$(ARCH)-$(TAG).aci: dist/flanneld-$(ARCH)-$(TAG).docker
+	docker2aci dist/flanneld-$(ARCH)-$(TAG).docker
+	mv quay.io-coreos-flannel$(IMAGE_SUFFIX)-$(TAG).aci dist/flanneld-$(ARCH)-$(TAG).aci
+	actool patch-manifest --replace --capability=CAP_NET_ADMIN \
+      --mounts=run-flannel,path=/run/flannel,readOnly=false:etc-ssl-etcd,path=/etc/ssl/etcd,readOnly=true:dev-net,path=/dev/net,readOnly=false \
+      dist/flanneld-$(ARCH)-$(TAG).aci
+
+docker-push: dist/flanneld-$(ARCH)-$(TAG).docker
+	docker push $(REGISTRY)/flannel$(IMAGE_SUFFIX):$(TAG)
+
+## Build an architecture specific static flanneld binary
+dist/flanneld-$(ARCH):
+    # Build for other platforms with ARCH=$ARCH make build
+    # valid values for $ARCH are amd64 amd arm64 and ppc64le
+	docker run -e CC=$(CC) -e GOARM=$(GOARM) -e GOARCH=$(ARCH) -it \
+	    -v ${PWD}:/go/src/github.com/coreos/flannel:ro \
+        -v ${PWD}/dist:/go/src/github.com/coreos/flannel/dist \
+	    gcr.io/google_containers/kube-cross:$(KUBE_CROSS_TAG) /bin/bash -c '\
+		cd /go/src/github.com/coreos/flannel && \
+		CGO_ENABLED=1 make -e dist/flanneld && \
+		mv dist/flanneld dist/flanneld-$(ARCH) && \
+		file dist/flanneld-$(ARCH) && \
+		chown -R $(shell id -u):$(shell id -u) dist'
+
+## Build an architecture specific iptables binary
+dist/iptables-$(ARCH):
+	docker run -e CC=$(CC) -e GOARM=$(GOARM) -e GOARCH=$(ARCH) -it \
+            -v ${PWD}:/go/src/github.com/coreos/flannel:ro \
+            -v ${PWD}/dist:/go/src/github.com/coreos/flannel/dist \
+            gcr.io/google_containers/kube-cross:$(KUBE_CROSS_TAG) /bin/bash -c '\
+            curl http://www.netfilter.org/projects/iptables/files/iptables-1.4.21.tar.bz2 | tar -jxv && \
```

Comment:
`curl -sSL` is more reliable I think. Also, it might be worth it to make the iptables version a variable